relay.go

```diff
@@ -281,6 +282,22 @@ func (r *Relayer) Relay(f *Frame) (shouldRelease bool, _ error) {
 	return r.handleCallReq(cr)
 }
 
+func (r *Relayer) updateCallReqContinueChecksum(f *Frame, cs Checksum) {
+	rbuf := typed.NewReadBuffer(f.SizedPayload())
+	rbuf.SkipBytes(1) // flags
+	rbuf.SkipBytes(1) // checksum type
+
+	checksumRef := typed.BytesRef(rbuf.ReadBytes(cs.Size()))
+
+	n := rbuf.ReadUint16()
+	if n == 0 {
+		return
```

Comment:
can we add a log here?

do we have any way to get coverage here?

Reply:
some discoveries/conclusions thanks to some offline pairing with @prashantv:
- The relay code will consider arg2 fragmented when it encounters a call without arg3. Thus it is nearly impossible to successfully relay a call where arg2 fills the entire `calReq` frame
- A `callReqContinue` frame with 0-length arg3 can be deliberately created by writing 0 bytes to arg3, then flushing before closing the writer. Although uncommon, it can still happen in practice
- The code above is incorrect in that it doesn't update the `callReqContinue` frame with the running checksum, which will result in inconsistent checksums for the call and fail that call

Since the following code correctly handles the case where len(arg3) == 0, we've simply removed the condition.